presto-main/src/main/java/com/facebook/presto/type/ArrayParametricType.java

```diff
@@ -40,7 +40,9 @@ public String getName()
     @Override
     public Type createType(List<TypeParameter> parameters)
     {
-        checkArgument(parameters.size() == 1, "Array type expects exactly one type as a parameter, got %s", parameters);
+        if (parameters.isEmpty()) {
```

Comment:
This seems wrong. CreateType should never be called if the provided arguments don't match the constructor of the parametric type. As an analogy, think about how we validate function calls for normal expressions: match the args and types, print any errors at that point, then execute the call at runtime